dspy/clients/lm.py

```diff
@@ -92,7 +92,7 @@ def __call__(self, prompt=None, messages=None, **kwargs):
             completion = cached_litellm_text_completion if cache else litellm_text_completion
 
         response = completion(
-            request=ujson.dumps(dict(model=self.model, messages=messages, **kwargs)),
```

Comment:
When response_format is a pydantic model (recommended by LiteLLM), `ujson.dumps()` fails because pydantic models are not directly serializable using `ujson.dumps()`. This line diff is a temporary hack to get the implementation working end-to-end for test purposes.

Reply:
Hi @dbczumar, please let me know if [this approach](https://gist.github.com/rohitgarud/80bdcd30b65c154e07f343055f95898e) is worthy of a PR, I am happy to contribute